Book-A-Desk.Api.Tests/EmailNotificationTests.fs

```diff
@@ -0,0 +1,63 @@
+module Book_A_Desk.Api.Tests.EmailNotificationTests
+
+open System
+open Book_A_Desk.Api
+open Book_A_Desk.Api.Models
+open Book_A_Desk.Domain.Office.Domain
+open Xunit
+
+[<Fact>]
+let ``GIVEN A booking WHEN calling to SendEmailNotification THEN Email must be sent`` () = async {
+    let officeId = Guid.NewGuid ()
+    let totalDesks = 32
+    let mockOffice =
+        {
+            Id = officeId |> OfficeId
+            City = CityName "SomeCityName"
+            BookableDesksPerDay = totalDesks
+        }
+    let mockedOfficeReference =
+        {
+            Id = officeId.ToString()
+        }
+    let offices =
+        mockOffice |> List.singleton
+    let mockGetOffices () = offices
+    
+    let date = DateTime(2021,02,01)
+    
+    let mockedUser =
+        {
+            Email = "booking.user@broadsing.com"    
+        }
+    
+    let mockBooking =
+        {
+            Office = mockedOfficeReference
+            Date = date
+            User = mockedUser
+        }
+        
+    let mockEmailConfig =
+        {
+            SmtpClientUrl = "http://localhost:8080/SMTP"
+            SmtpUsername = "username"
+            SmtpPassword = "password"
+            EmailSender = "from@broadsing.com"
+            EmailReviewer = "reviewer@broadsing.com"
+        }
+        
+    let mockEmailServiceConfiguration () = mockEmailConfig
+            
+    let emailNotification = EmailNotification.initialize mockEmailServiceConfiguration mockGetOffices       
+    
+    // Refactor this code to mock SmtpClient and verify that Send function is called
+    try
+        emailNotification.SendEmailNotification mockBooking
```

Comment:
Opinion:

Is this code worth testing? I recommend not to test infrastructure code in unit tests for 2 reasons:
1. We should allow infrastructure to change. When you unit tests, you add a coupling tests <=> code. This coupling makes it harder to change infrastructure
2. Infrastructure code should be really simple. There should not be any logic, it should call an external service with no know values and end there. The only worthwhile test then becomes "makes sure it calls the external service", which is a trivially simple and not worth it test. If it's more complicated than that, it usually means that we mixed in business logic with it.

I think this is the case here.